CHANGES.md

```diff
@@ -1,6 +1,48 @@
 Release Changes List
 ====================
 
+version 1.24.1
+==============
+
+Update to the twenty-seventh public release of Chapel, April 2021  
+(see also changes below for 1.24.0)
+
+Highlights
+----------
+* significant performance improvements for InfiniBand systems
+* improved support for computing with `enum` ranges
+* closed all known memory leaks
+
+Feature Improvements
+--------------------
+* extended `param` for-loops to support `enum` ranges
+* added support for open-interval `enum` ranges
+
+Performance Optimizations / Improvements
+----------------------------------------
+* improved NUMA affinity and startup times when using a fixed heap
+* improved our approach to polling when communicating with 'gasnet/ucx'
+
+Memory Improvements
+-------------------
+* closed a memory leak related to tuple casts
```

Comment:
Casting the tuple was the workaround and not the problem. A more accurate characterization would be `... related to tuple coercions`

More: the leak happened when you pass a tuple of shared non-nilables into a function that expects tuple of shared nilables. If you didn't explicitly cast, the compiler would do an implicit conversion between those two types, and that caused the leak, because that implicit conversion was not recognized as an initialization that we need to clean up after.